oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2AuthorizationCodeAuthenticationProvider.java

```diff
@@ -149,4 +145,9 @@ public Authentication authenticate(Authentication authentication) throws Authent
 	public boolean supports(Class<?> authentication) {
 		return OAuth2AuthorizationCodeAuthenticationToken.class.isAssignableFrom(authentication);
 	}
+
+	private boolean isConfiguredForRefreshToken(RegisteredClient registeredClient) {
+		return registeredClient.getTokenSettings().enableRefreshTokens() &&
+				registeredClient.getAuthorizationGrantTypes().contains(AuthorizationGrantType.REFRESH_TOKEN);
```

Comment:
Just my 2 cents here since I was working on the original PR to support refresh token grant.  

I somehow missed this supported authorization grant types. And now I'm not even sure now how these two settings can be separated. It almost feels like `enableRefreshToken` flag can be removed completely and if the grant type is supported then refresh tokens should be issued.

Reply:
Yes. I agree. I don't see a reason for that flag. Should I then remove it completely? I think I need some second opinion here before I do that change.

Reply:
I agree as well. Let's remove `enableRefreshToken()` and simply check `registeredClient.getAuthorizationGrantTypes().contains(AuthorizationGrantType.REFRESH_TOKEN)`.